File: FitToFit/FitToFit.Services/Database/Rezervacije.cs
using System;
using System.Collections.Generic;

namespace FitToFit.Services.Database;

public partial class Rezervacije
{
    public int RezervacijaId { get; set; }

    public DateTime Datum { get; set; }

    public int KorisnikId { get; set; }

    public int ClanarinaId { get; set; }

    public string? StateMachine { get; set; }

    public decimal? Iznos { get; set; }

    public DateTime? DatumIsteka { get; set; }

    public string? BrojTransakcije { get; set; }

    public virtual Clanarine Clanarina { get; set; } = null!;

    public virtual Korisnici Korisnik { get; set; } = null!;

    public virtual ICollection<RezervacijaStavke> RezervacijaStavkes { get; set; } = new List<RezervacijaStavke>();
}

File: FitToFit/UI/fittofit_mobile/lib/models/rezervacije.dart
import 'package:fittofit_mobile/models/rezervacijaStavke.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rezervacije.g.dart';

@JsonSerializable()
class Rezervacije {
  int rezervacijaId;
  DateTime datum;
  String? stateMachine;
  double? iznos;
  int korisnikId;
  int clanarinaId;
  DateTime? datumIsteka;
  String? brojTransakcije;
  List<RezervacijaStavke>? rezervacijaStavkes = [];

  Rezervacije(
      {required this.rezervacijaId,
      required this.datum,
      this.stateMachine,
      this.iznos,
      required this.korisnikId,
      required this.clanarinaId,
      this.rezervacijaStavkes,
      this.datumIsteka,
      this.brojTransakcije});

  factory Rezervacije.fromJson(Map<String, dynamic> json) =>
      _$RezervacijeFromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijeToJson(this);
}
